### PR TITLE
Refactor filter matching

### PR DESCRIPTION
**Pull request by yanivagman**
### 1. Explain what the PR does

"Replace me with `make check-pr` output"

### CODE REVIEW

**Review by geyslan — APPROVED** [review]
LGTM

File: pkg/ebpf/c/types.h [hunk 1]
```diff
@@ -294,43 +294,43 @@ typedef struct ksym_name {
...
+    u64 uts_ns_filter_enabled;
+    u64 comm_filter_enabled;
+    u64 cgroup_id_filter_enabled;
+    u64 cont_filter_enabled;
+    u64 new_cont_filter_enabled;
+    u64 new_pid_filter_enabled;
+    u64 proc_tree_filter_enabled;
+    u64 bin_path_filter_enabled;
+    u64 follow_filter_enabled;
+    // bitmap indicating the default match behavior for policies
```
Comment (geyslan) [inline pkg/ebpf/c/types.h:319]: What about explaining more `*_filter_default_match`? The enabled above is crystal, but default_match not so much.

File: pkg/ebpf/c/types.h [hunk 2]
```diff
@@ -294,43 +294,43 @@ typedef struct ksym_name {
...
 typedef struct equality {
-    // bitmask with scopes on which a equal '=' filter is set
-    // its bit value will depend on the filter's equality precedence order
-    u64 equal_in_scopes;
-    // bitmask with scopes on which a filter equality is set
-    u64 equality_set_in_scopes;
+    // bitmap indicating which policies have a scope filter that uses the '=' operator (0 means
+    // '!=')
+    u64 equals_in_policies;
+    // bitmap indicating which policies have a scope filter that utilize the provided key
```
Comment (rscampos) [inline pkg/ebpf/c/types.h:300]: Same here with word "scope".

File: pkg/ebpf/c/types.h (hunk 1 above)
Comment (rscampos) [reply pkg/ebpf/c/types.h:319]: good point @geyslan , maybe something like this if I understood correctly: `default behavior to match certain events even if no explicit key match occurs`

**Review by rscampos — APPROVED** [review]
LGTM

**Pull request merged**